shared/chat/inbox/row/bottom-line.js

```diff
@@ -0,0 +1,140 @@
+// @flow
+import React from 'react'
+import {Text, Markdown, Box} from '../../../common-adapters'
+import {
+  globalStyles,
+  globalColors,
+  globalMargins,
+  styleSheetCreate,
+  collapseStyles,
+  lineHeight,
+} from '../../../styles'
+import {isMobile} from '../../../constants/platform'
+
+type BottomLineProps = {
+  backgroundColor: ?string,
+  participantNeedToRekey: boolean,
+  showBold: boolean,
+  snippet: ?string,
+  subColor: ?string,
+  youNeedToRekey: boolean,
+}
+
+function BottomLine(props: BottomLineProps) {
```

Comment:
Converted everything back. Can you explain a bit more about why functional components aren't as optimized in this case? Is it just a general thing that inheriting from PureComponent does the right thing whereas functional components don't?